### ISSUE TITLE
Why does the field of path under permissions look like so weird

### ISSUE THREAD

**Issue by YvesZHI** [envoyproxy/envoy#25731]
I'm using Envoy as the gateway of my backend. I want to use two filters: JWT and RBAC.

What I'm trying to do is to add the role info into the payload of JWT. When a request is sent to Envoy, Envoy could validate the JWT and extract the payload. So now Envoy has had the role info of the request. Then, the role info can be used in the RBAC filter to verify if the request has the permission to access the cluster.

Here is an example of the payload in JWT:
```
payload = {
    'iss': 'ImIss',
    'aud': 'backend_service',
    'role': 'admin',    # admin or guest
    'exp': datetime.datetime.utcnow() + datetime.timedelta(minutes=300)
}
```

I used `payload_in_metadata` to store the payload of JWT into a namesapce and tried to extract it in the RBAC filter. Here is my config:
```
static_resources:
  listeners:
  - name: listener_0
    address:
      socket_address:
        address: 0.0.0.0
        port_value: 9989
    filter_chains:
      - filters:
          - name: envoy.filters.network.http_connection_manager
            typed_config:
              "@type": type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager
              stat_prefix: ingress_http
              http_filters:
                - name: envoy.filters.http.jwt_authn
                  typed_config:
                    "@type": type.googleapis.com/envoy.extensions.filters.http.jwt_authn.v3.JwtAuthentication
                    providers:
                      remote_jwt_provider:
                        issuer: "ImIss"
                        audiences: ["backend_service"]
                        remote_jwks:
                          http_uri:
                            uri: http://172.12.1.44:9081/jwts.json
                            cluster: auth_service
                            timeout: 3s
                          cache_duration:
                            seconds: 300
                        forward: true
                        payload_in_metadata: my_payload
                        from_params:
                          - jwt_token
                    rules:
                      - match:
                          prefix: "/"
                        requires:
                          provider_name: remote_jwt_provider
                - name: envoy.filters.http.rbac
                  typed_config:
                    "@type": type.googleapis.com/envoy.extensions.filters.http.rbac.v3.RBAC
                    rules:
                      action: ALLOW
                      policies:
                        "admin":
                          permissions:
                            - any: true
                          principals:
#                            - any: true
                            - metadata:
                                filter: envoy.filters.http.jwt_authn
                                path:
                                  - key: my_payload
                                  - key: role
                                value:
                                  string_match:
                                    exact: "admin"
                - name: envoy.filters.http.router
                  typed_config:
                    "@type": type.googleapis.com/envoy.extensions.filters.http.router.v3.Router
              route_config:
                name: local_route
                virtual_hosts:
                  - name: local_service
                    domains: ["*"]
                    routes:
                      - match: { prefix: "/" }
                        route:
                          cluster: cluster_example
```

Well, it seems to work but I'm so confused by the field `path` under `permissions`.

As my understanding, the payload in the Envoy should look like this:
```
  my_payload:
    'iss': 'ImIss',
    'aud': 'backend_service',
    'role': 'admin',    # admin or guest
```
In other words, `my_payload` and `role` are not at the same level. But 
```
path:
  - key: my_payload
  - key: role
```
The content of the field `path` is a list, so all of elements of the list look like at the same level.

I'm confused. Why is RBAC of Envoy designed like this?

**Comment by phlax** [comment]
@yangminzhu @kyessenov any ideas about this ?
